### PR TITLE
Add handling for pre and code blocks

### PR DESCRIPTION
**Pull request by jongracecox**
* `<pre>` and `<code>` blocks use Courier font
* `<pre>` blocks retain white space and new line characters.

Add manual test for code blocks (similar to the tables test). Run this manually to validate visual representation of code and pre-formatted blocks.

Remove calls to `remove_whitespace()` from `run_process()`. This is now called from `handle_data()` when not in a pre block.

Test HTML:

```
<h1>Code Test</h1>

<p>This test validates handling of code blocks.</p>

<h2>Pre-formatted block</h2>

<pre>
This is a pre-formatted block.
  That should be pre-formatted.
Retaining any carriage returns, and     all    white    space.

And blank lines.
</pre>

<h2>Code block</h2>

<p><code>
This is a code block.
  That should be NOT be pre-formatted.
It should NOT retain carriage returns, or     all    white    space.

or blank lines.
</code></p>

<h2>Code elements</h2>

<p>This is a sentence that includes <code>code</code> elements.</p>
```

In HTML, this renders like this:

> image

The resulting Word doc looks like this:

> image

### CODE REVIEW

File: tests/test_code.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import os
```
Comment (pqzx) [inline tests/test_code.py:1]: Would be good to include this test in the workflow somehow. E.g. could add [here](https://github.com/pqzx/html2docx/blob/b9d30ceca5da5cc6e398b03832f1dd3448e8b3aa/.github/workflows/python-app.yml#L35) (and include the artifact in test outputs).

**Review by pqzx — CHANGES_REQUESTED** [review]
Looks great. Minor comment re: testing.

File: tests/test_code.py (hunk 1 above)
Comment (jongracecox) [reply tests/test_code.py:1]: That's a good idea. I'll try and add that over the next few days.

Comment (jongracecox) [reply tests/test_code.py:1]: I ended up making a few changes:

- Add unittests to test code block, pre block and inline use of code block.
- Made a change to `remove_whitespace` when used for code blocks. This prevents a space character being inserted at the beginning of code blocks.
- Added doctests to demonstrate how `remove_whitespace` works. This is more for clarity than anything, but minor changes to the function could cause big changes to how things are rendered. Adding doctests and running them in the workflow will ensure any changes don't break anything.

I still need to update the workflow ([can't do it over an https clone?!](https://stackoverflow.com/a/58885125/6252525)), but I'm planning to add this to run the `test_code` test module and the doctests in `h2d`.

```
        python -m tests.test_code
        python -m doctest -v htmldocx/h2d.py
```

The outputs end up looking like this:

> image

> image

> image

Note the difference from the previous version that ended up with the extra white space:

> image

Comment (pqzx) [reply tests/test_code.py:1]: Great work on this. Thanks for the clear explanation 😄

**Review by pqzx — APPROVED** [review]

**Pull request merged**